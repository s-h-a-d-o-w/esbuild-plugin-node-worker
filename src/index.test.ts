import esbuild from "esbuild";
import { existsSync } from "node:fs";
import { mkdir, readFile, rm } from "node:fs/promises";
import path from "node:path";
import { beforeEach, describe, expect, it } from "vitest";

import { nodeWorker } from "./index.js";

const outdir = path.join(import.meta.dirname, "__test-output__");
const fixturesDir = path.join(import.meta.dirname, "__fixtures__");

async function matchSnapshot(relativePath: string) {
  const content = await readFile(path.join(outdir, relativePath), "utf8");
  expect(content).toMatchSnapshot();
}

describe("esbuild-plugin-node-worker", () => {
  beforeEach(async () => {
    if (existsSync(outdir)) {
      await rm(outdir, { recursive: true });
    }
    await mkdir(outdir, { recursive: true });
  });

  it("ESM works", async () => {
    await esbuild.build({
      entryPoints: [path.join(fixturesDir, "esm/main.js")],
      outdir,
      format: "esm",
      plugins: [nodeWorker],
    });

    await matchSnapshot("main.js");
    await matchSnapshot("worker.mjs");
    await matchSnapshot("nested/nested-worker.mjs");
  });

  it("CJS works", async () => {
    await esbuild.build({
      entryPoints: [path.join(fixturesDir, "cjs/main.js")],
      outdir,
      format: "cjs",
      plugins: [nodeWorker],
    });

    await matchSnapshot("main.js");
    await matchSnapshot("worker.cjs");
    await matchSnapshot("nested-worker.cjs");
  });
});
