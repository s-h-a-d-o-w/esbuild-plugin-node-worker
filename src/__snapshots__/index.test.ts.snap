// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild-plugin-node-worker > CJS works 1`] = `
""use strict";
const { Worker } = require("worker_threads");
new Worker(require("path").join(__dirname, "./worker.cjs"), {
  workerData: "foo"
});
"
`;

exports[`esbuild-plugin-node-worker > CJS works 2`] = `
""use strict";
const path = require("path");
const { Worker } = require("worker_threads");
class NestedWorker extends Worker {
  constructor(path2, options) {
    super(path2, options);
  }
}
const nestedWorker = new NestedWorker(require("path").join(__dirname, "./nested-worker.cjs"), {
  workerData: "foo"
});
const nestedWorker2 = new Worker(require("path").join(__dirname, "nested-worker.cjs"), {
  workerData: "foo"
});
console.log("Worker says hi");
"
`;

exports[`esbuild-plugin-node-worker > CJS works 3`] = `
""use strict";
console.log("Nested worker says hi");
"
`;

exports[`esbuild-plugin-node-worker > ESM works 1`] = `
"import { Worker } from "node:worker_threads";
new Worker(new URL("./worker.mjs", import.meta.url), {
  workerData: "foo"
});
"
`;

exports[`esbuild-plugin-node-worker > ESM works 2`] = `
"import { Worker } from "node:worker_threads";
class NestedWorker extends Worker {
  constructor(path2, options) {
    super(path2, options);
  }
}
const nestedWorker = new NestedWorker(new URL("./nested/nested-worker.mjs", import.meta.url), {
  workerData: "foo"
});
const nestedWorker2 = new Worker(new URL("nested/nested-worker.mjs", import.meta.url), {
  workerData: "foo"
});
console.log("Worker says hi");
"
`;

exports[`esbuild-plugin-node-worker > ESM works 3`] = `
"console.log("Nested worker says hi");
"
`;
